test_conformance/extensions/cl_khr_command_buffer/negative_command_nd_range_kernel.cpp

```diff
@@ -0,0 +1,525 @@
+                               "CL_INVALID_CONTEXT",
+                               TEST_FAIL);
+
+        return CL_SUCCESS;
+    }
+
+    cl_int CreateKernelWithDifferentContext()
+    {
+        cl_int error = CL_SUCCESS;
+
+        new_context = clCreateContext(0, 1, &device, nullptr, nullptr, &error);
+        test_error(error, "Failed to create context");
+
+        const char* kernel_str =
+            R"(
+      __kernel void copy(__global int* in, __global int* out, __global int* offset) {
+          size_t id = get_global_id(0);
+          int ind = offset[0] + id;
+          out[ind] = in[ind];
+      })";
+
+        error = create_single_kernel_helper_create_program(
+            new_context, &program, 1, &kernel_str);
+        test_error(error, "Failed to create program with source");
+
+        error = clBuildProgram(program, 1, &device, nullptr, nullptr, nullptr);
+        test_error(error, "Failed to build program");
+
+        kernel = clCreateKernel(program, "copy", &error);
+        test_error(error, "Failed to create copy kernel");
+
+        return CL_SUCCESS;
+    }
+    clContextWrapper new_context;
+};
+
+// CL_INVALID_SYNC_POINT_WAIT_LIST_KHR if sync_point_wait_list is NULL and
+// num_sync_points_in_wait_list is > 0, or sync_point_wait_list is not NULL and
+// num_sync_points_in_wait_list is 0, or if synchronization-point objects in
+// sync_point_wait_list are not valid synchronization-points.
+struct CommandNDRangeKerneSyncPointsNullOrNumZero
+    : public BasicCommandBufferTest
+{
+    using BasicCommandBufferTest::BasicCommandBufferTest;
+
+    cl_int Run() override
+    {
+        cl_int error = clCommandNDRangeKernelKHR(
+            command_buffer, nullptr, nullptr, kernel, 1, nullptr, &num_elements,
+            nullptr, 1, nullptr, nullptr, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_SYNC_POINT_WAIT_LIST_KHR,
+                               "clCommandNDRangeKernelKHR should return "
+                               "CL_INVALID_SYNC_POINT_WAIT_LIST_KHR",
+                               TEST_FAIL);
+
+        cl_sync_point_khr point = 1;
+        cl_sync_point_khr* sync_points[] = { &point };
+        error = clCommandNDRangeKernelKHR(
+            command_buffer, nullptr, nullptr, kernel, 0, nullptr, &num_elements,
+            nullptr, 0, sync_points[0], nullptr, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_SYNC_POINT_WAIT_LIST_KHR,
+                               "clCommandNDRangeKernelKHR should return "
+                               "CL_INVALID_SYNC_POINT_WAIT_LIST_KHR",
+                               TEST_FAIL);
+
+        cl_sync_point_khr* invalid_point = nullptr;
+        cl_sync_point_khr* invalid_sync_points[] = { invalid_point };
+        error = clCommandNDRangeKernelKHR(
+            command_buffer, nullptr, nullptr, kernel, 0, nullptr, &num_elements,
+            nullptr, 0, invalid_sync_points[0], nullptr, nullptr);
```

Comment:
This seems overly complex, `invalid_sync_points[0]` is a `nullptr` - we could just pass a `nullptr` literal here rather than defining `invalid_point` or `invalid_sync_points`.

Doing so would show that passing 0 for `num_sync_points_in_wait_list` and `nullptr` for `sync_point_wait_list` is valid. Since we haven't recorded any commands, at this point there shouldn't be any valid sync-point values and I think you
could just do
```cpp
cl_sync_point_khr invalid_point = 0;  // No sync points defined yet, so any valid should be invalid`
```